### PR TITLE
Implementing db

### PR DESCRIPTION
**Pull request by aSvetenco**
in this PR 
implementing ROOM db

### CODE REVIEW

File: MeetUp/app/src/main/java/com/endava/meetup/MeetUpApp.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+    private static Context context;
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        context = getApplicationContext();
+    }
+
+    public static Context getAppContext() {
```
Comment (CebotariM) [inline MeetUp/app/src/main/java/com/endava/meetup/MeetUpApp.java]: bad practice, where do you need the context?

File: MeetUp/app/src/main/java/com/endava/meetup/data/db/room/MeetUpDbFactory.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    private static MeetUpDbFactory factory;
+    private MeetUpDb meetUpDb;
+
+    private MeetUpDbFactory() {
+        buildDbInstance();
+    }
+
+    public static MeetUpDbFactory getInstance() {
+        if(factory == null) {
+            factory = new MeetUpDbFactory();
```
Comment (CebotariM) [inline MeetUp/app/src/main/java/com/endava/meetup/data/db/room/MeetUpDbFactory.java]: do you need a static instance of the factory?

File: MeetUp/app/src/main/java/com/endava/meetup/screens/home/HomeActivityModel.java [hunk 3]
```diff
@@ -55,7 +62,12 @@ public void fetchData() {
 
     @Override
     public void onError(Throwable throwable) {
-        callback.onError(errorHandler.handleError(throwable));
+        String errorMessage = errorHandler.handleError(throwable);
+        if (throwable instanceof NoNetworkException) {
+            tryToUseDbData(errorMessage);
```
Comment (CebotariM) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/home/HomeActivityModel.java]: why shouldn't we try to use DB data on any error?

File: MeetUp/app/src/test/java/com/endava/meetup/screens/home/HomeActivityModelTest.java [hunk 4]
```diff
@@ -138,6 +138,40 @@ public void onErrorWithResponseThrowable_ShouldCallOnError() {
         Mockito.verify(callback).onError(errorHandler.handleError(throwable));
     }
 
+    @Test
+    public void onErrorWithResponseThrowable_ShouldTryToGetDataFromDb_WhenNoNetworkExceptionOccur() {
```
Comment (CebotariM) [inline MeetUp/app/src/test/java/com/endava/meetup/screens/home/HomeActivityModelTest.java]: you can omit the "_", it's ok to have a method name this long, as long as it explains what the method does

File: MeetUp/app/src/main/java/com/endava/meetup/MeetUpApp.java (hunk 1 above)
Comment (aSvetenco) [reply MeetUp/app/src/main/java/com/endava/meetup/MeetUpApp.java]: I need appContext for initializing Database

File: MeetUp/app/src/main/java/com/endava/meetup/data/db/room/MeetUpDbFactory.java (hunk 2 above)
Comment (aSvetenco) [reply MeetUp/app/src/main/java/com/endava/meetup/data/db/room/MeetUpDbFactory.java]: I used the singleton for being sure that app has only one instance of DataBase. Maybe factory isn't the best variable name...

File: MeetUp/app/src/main/java/com/endava/meetup/screens/home/HomeActivityModel.java (hunk 3 above)
Comment (aSvetenco) [reply MeetUp/app/src/main/java/com/endava/meetup/screens/home/HomeActivityModel.java]: If we don't have the network connection, app checks database and if there are stored data - shows for user.

File: MeetUp/app/src/test/java/com/endava/meetup/screens/home/HomeActivityModelTest.java (hunk 4 above)
Comment (aSvetenco) [reply MeetUp/app/src/test/java/com/endava/meetup/screens/home/HomeActivityModelTest.java]: ok, I used them for separate methods name, behavior and conditions

File: MeetUp/app/src/main/java/com/endava/meetup/data/db/room/MeetUpDbFactory.java (hunk 2 above)
Comment (CebotariM) [reply MeetUp/app/src/main/java/com/endava/meetup/data/db/room/MeetUpDbFactory.java]: you need a static DB, not a static factory

File: MeetUp/app/src/main/java/com/endava/meetup/screens/home/HomeActivityModel.java (hunk 3 above)
Comment (CebotariM) [reply MeetUp/app/src/main/java/com/endava/meetup/screens/home/HomeActivityModel.java]: if (throwable instanceof NoNetworkException) {...} else -> why here you shouldn't call DB? what if it's not a NoNetworkException, but some other type of exception?

**Pull request merged**